src/net/sourceforge/kolmafia/persistence/SkillDatabase.java

```diff
@@ -1532,28 +1532,28 @@ public static final boolean contains(final String skillName) {
   }
 
   public static final void generateSkillList(final StringBuffer buffer, final boolean appendHTML) {
-    ArrayList<String>[] categories = new ArrayList[SkillDatabase.CATEGORIES.length];
+    List<ArrayList<String>> categories = new ArrayList<>(SkillDatabase.CATEGORIES.length);
```

Comment:
Can this just be `List<List<` or is the fact that its `ArrayList` important?

Reply:
SkillDatabase.appendSkillList(StringBuffer, boolean, String, ArrayList<String>) is called at the end

Reply:
(so yes, it's important to note that they are ArrayLists)

Reply:
But does *that* need to specifically be an ArrayList?

Reply:
Uhhhhh... no!

Reply:
Uhhhhh... no!